Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -178,6 +185,48 @@ actor ExternalBuildSystemAdapter {
     ).connection
   }
 
+  private static func getConfigPath(for workspaceFolder: AbsolutePath? = nil) -> AbsolutePath? {
+    var buildServerConfigLocations: [AbsolutePath?] = []
+    if let workspaceFolder = workspaceFolder {
+      buildServerConfigLocations.append(workspaceFolder.appending(component: ".bsp"))
+    }
+
+    #if os(Windows)
+    let localAppDataPath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["LOCALAPPDATA"])?.appending(component: "bsp")
+    let programDataPath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["PROGRAMDATA"])?.appending(component: "bsp")
+
+    buildServerConfigLocations.append(contentsOf: [localAppDataPath, programDataPath])
+    #else
+    if let libraryUrl = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first {
+      let libraryPath = AbsolutePath(validatingOrNil: libraryUrl.absoluteString)?.appending(component: "bsp")
+      buildServerConfigLocations.append(libraryPath)
+    }
+
+    let xdgDataHomePath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["XDG_DATA_HOME"])?.appending(component: "bsp")
```

Comment:
```suggestion
    let xdgDataHomePath = AbsolutePath(validatingOrNil: ProcessInfo.processInfo.environment["XDG_DATA_HOME"])?.appending(component: "bsp")

    if let libraryUrl = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first {
      let libraryPath = AbsolutePath(validatingOrNil: libraryUrl.absoluteString)?.appending(component: "bsp")
      buildServerConfigLocations.append(libraryPath)
    }
```

Should be the other way round to follow the spec.